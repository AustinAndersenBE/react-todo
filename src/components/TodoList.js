import { useSelector } from 'react-redux';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

function TodoList() {
  const todos = useSelector((state) => state.todos);

  return (
    <div>
      <NewTodoForm />
      {todos.map((todo) => (
        <Todo key={todo.id} id={todo.id} task={todo.task} />
      ))}
    </div>
  );
}

export default TodoList;