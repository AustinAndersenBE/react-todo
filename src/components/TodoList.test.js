import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import TodoList from './TodoList';

const mockStore = configureMockStore();
const store = mockStore({
  todos: [
    { id: '1', task: 'Test task 1' },
    { id: '2', task: 'Test task 2' },
  ],
});

//Smoke Test
test('renders without crashing', () => {
  render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );
  const formElement = screen.getByText("Add Todo");
  expect(formElement).toBeInTheDocument();
});

//Snapshot Test
test('matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

//Business Logic Test
test('renders todos from state', () => {
  render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );

  // Check if the todos from state are rendered
  const todo1 = screen.getByText('Test task 1');
  const todo2 = screen.getByText('Test task 2');
  expect(todo1).toBeInTheDocument();
  expect(todo2).toBeInTheDocument();
})