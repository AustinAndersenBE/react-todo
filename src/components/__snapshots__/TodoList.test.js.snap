// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <div>
      <span>
        Test task 1
      </span>
      <button>
        X
      </button>
    </div>
    <div>
      <span>
        Test task 2
      </span>
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
